source/string.lua

```diff
@@ -0,0 +1,153 @@
+--[[
+    derived from documentation and reference implementation at:
+    https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf
+    Attributions and copyright licensing by Mozilla Contributors is licensed under CC-BY-SA 2.5
+]]
+--!strict
+local Array = require(script.Parent.array)
+type Array<T> = Array.Array<T>
+local RegExp = require(script.Parent.regex)
+type RegExp = RegExp.RegExp
+local exports = {}
+
+local NaN = 0 / 0
+-- TODO?: support utf8
+exports.charCodeAt = function(str: string, index: number): number
+  if index < 1 or index >= string.len(str) then
+    return NaN
+  end
+  local result = string.byte(str, index)
+  return if result == nil then NaN else result
+end
+
+exports.lastIndexOf = function(str: string, findValue: string, _fromIndex: number?): number
```

Comment:
Looks like `_fromIndex` isn't supposed to be ignored. Maybe this would be better named as `fromIndex_`?